### ISSUE TITLE
Limit Being Applied to Include when using SingleOrDefault

### ISSUE THREAD

**Issue by kluhman** [JasperFx/marten#2258]
**Marten Version:** 5.5.1
I have a data structure that looks similar to the following

```csharp
public class User
{
    public Guid Id { get; set; }
    public List<Guid> TenantIds { get; set; }
}

public class Tenant
{
    public Guid Id { get; set; }
}
```

When I run a query to get a single user and all the tenants they belong to as follows, I get an incomplete set of results, specifically only 2 tenants instead of the whole list.

```csharp
var tenants = new Dictionary<Guid, Tenant>();
var user = _session
    .Query<User>()
    .Include<Tenant>(x => x.TenantIds, tenants)
    .SingleOrDefault(x => x.Id == id);
```

The SQL being generated is as follows:
```sql
drop table if exists mt_temp_id_list1;
create temp table mt_temp_id_list1 as (select id, d.id as id1, include2.id2
                                       from public.mt_doc_user as d
                                                LEFT JOIN LATERAL unnest(CAST(ARRAY(SELECT jsonb_array_elements_text(CAST(d.data ->> 'TenantIds' as jsonb))) as uuid[])) WITH ORDINALITY as include2(id2)
                                                          ON TRUE
                                       where (d.tenant_id = :tenantid and d.id = :p1)
                                       LIMIT :p2);
select d.data, d.mt_last_modified
from public.mt_doc_tenant as d
where (d.tenant_id = :tenantid and id in (select id2 from mt_temp_id_list1));
select d.data
from public.mt_doc_user as d
where id in (select id from mt_temp_id_list1)
LIMIT :p3
```

I confirmed this was working as expected on the previous version I was running, 5.3.1. I'm assuming this is a result of the changes made in #2225

**Comment by jeremydmiller** [comment]
@kluhman What's the priority of this for you? I'm digging into this, and a couple other related issues today and it's a little too much for me today. I think I want to back up and try this again late next week after I get caught up on other things.

**Comment by kluhman** [comment]
@jeremydmiller It's hard blocking us from upgrading past `5.3.1`, but there's nothing in `5.4.0` or `5.5.0` that we really need. So I wouldn't say it's an urgent priority, but we'll need it soon.

**Comment by jeremydmiller** [comment]
I worked on this a little bit again today, and this is going to take more time. It's a snarl of permutations.

**Comment by jeremydmiller** [comment]
Closed by #2284
